proxy/src/main/java/com/wavefront/agent/listeners/tracing/TracePortUnificationHandler.java

```diff
@@ -199,6 +197,76 @@ public static void preprocessAndHandleSpan(
     }
   }
 
+  public static void handleSpanLogs(
+      String message, ReportableEntityDecoder<JsonNode, SpanLogs> spanLogsDecoder,
+      ReportableEntityDecoder<String, Span> spanDecoder,
+      ReportableEntityHandler<SpanLogs, String> handler,
+      @Nullable Supplier<ReportableEntityPreprocessor> preprocessorSupplier,
+      @Nullable ChannelHandlerContext ctx, boolean alwaysSampleErrors,
+      Function<Span, Boolean> samplerFunc) {
+    List<SpanLogs> spanLogsOutput = new ArrayList<>(1);
+    try {
+      spanLogsDecoder.decode(JSON_PARSER.readTree(message), spanLogsOutput, "dummy");
+    } catch (Exception e) {
+      handler.reject(message, formatErrorMessage(message, e, ctx));
+      return;
+    }
+
+    for (SpanLogs spanLogs : spanLogsOutput) {
+      String spanMessage = spanLogs.getSpan();
+      if (spanMessage == null) {
+        // For backwards compatibility, report the span logs if span line data is not included
+        handler.report(spanLogs);
+      } else {
+        ReportableEntityPreprocessor preprocessor = preprocessorSupplier == null ?
+            null : preprocessorSupplier.get();
+        String[] spanMessageHolder = new String[1];
+
+        // transform the line if needed
+        if (preprocessor != null) {
+          spanMessage = preprocessor.forPointLine().transform(spanMessage);
+
+          if (!preprocessor.forPointLine().filter(message, spanMessageHolder)) {
+            if (spanMessageHolder[0] != null) {
+              handler.reject(spanLogs, spanMessageHolder[0]);
+            } else {
+              handler.block(spanLogs);
+            }
+            return;
+          }
+        }
+        List<Span> spanOutput = new ArrayList<>(1);
+        try {
+          spanDecoder.decode(spanMessage, spanOutput, "dummy");
```

Comment:
`dummy`?

Reply:
it doesn't really matter in this context, but we should probably standardize on something like "default" or "unknown"

Reply:
yeah, right now i think we're defaulting to "dummy" in most places in the code. let me know if i should change that